packages/nextjs/hooks/types.ts

```diff
@@ -0,0 +1,28 @@
+export type TimelineEvent =
+  | {
+      type: "on-chain-checkin";
+      date: string;
+      title: string;
+      description: string;
+      address?: string;
+      link?: string;
+      author?: never;
+    }
+  | {
+      type: "graduation-nft";
+      date: string;
+      title: string;
+      description: string;
+      address?: string;
+      link?: string;
+      author?: never;
+    }
+  | {
+      type: "pr-merged";
+      date: string;
+      title: string;
+      description: string;
+      author?: string;
+      address?: never;
+      link?: string;
+    };
```

Comment:
This is very repetitive, use a base type, something like this:
```suggestion
type BaseTimelineEvent = {
      date: string;
      title: string;
      description: string;
      link?: string;
    }
type VariantTimelineEvent =
  | {
      type: "on-chain-checkin";
      address?: string;
      author?: never;
    }
  | {
      type: "graduation-nft";
      address?: string;
      author?: never;
    }
  | {
      type: "pr-merged";
      author?: string;
      address?: never;
    };
export type TimelineEvent = VariantTimelineEvent & BaseTimelineEvent;
```